src/components/AppProvider/utilities/withAppProvider/withAppProvider.tsx

```diff
@@ -1,94 +1,114 @@
-    theme: ThemeContext;
-    appBridge: ClientApplication<{}>;
-    subscribe(callback: () => void): void;
-    unsubscribe(callback: () => void): void;
+    theme?: ThemeProviderContext;
+    appBridge?: ClientApplication<{}>;
   };
 }
 
-export default function withAppProvider<OwnProps>() {
-  return function addProvider<C>(
-    WrappedComponent:
-      | React.ComponentClass<OwnProps & WithAppProviderProps> & C
-      | React.SFC<OwnProps & WithAppProviderProps> & C,
-  ): React.ComponentClass<OwnProps> & C {
-    class WithProvider extends React.Component<OwnProps, never> {
-      static contextTypes = WrappedComponent.contextTypes
-        ? merge(WrappedComponent.contextTypes, polarisAppProviderContextTypes)
-        : polarisAppProviderContextTypes;
+export interface Options {
+  inScrollable?: boolean;
+}
 
-      componentDidMount() {
-        const {
-          polaris: {subscribe: subscribeToPolaris},
-        } = this.context;
+function withScrollable<P, T>(WrappedComponent: ReactComponent<P, T>) {
+  class WithScrollable extends React.Component {
+    static contextTypes = WrappedComponent.contextTypes;
+    private stickyManager: StickyManager = new StickyManager();
 
-        if (subscribeToPolaris) {
-          subscribeToPolaris(this.handleContextUpdate);
-        }
-      }
+    render() {
+      return (
+        <AppProviderContext.Consumer>
+          {(polaris) => (
+            <AppProviderContext.Provider
+              value={{
+                ...polaris,
+                stickyManager: this.stickyManager,
+              }}
+            >
+              <WrappedComponent {...this.props} />
+            </AppProviderContext.Provider>
+          )}
+        </AppProviderContext.Consumer>
+      );
+    }
+  }
 
-      componentWillUnmount() {
-        const {
-          polaris: {unsubscribe: unsubscribeToPolaris},
-        } = this.context;
+  return WithScrollable;
+}
 
-        if (unsubscribeToPolaris) {
-          unsubscribeToPolaris(this.handleContextUpdate);
-        }
-      }
+export default function withAppProvider<OwnProps>({
+  inScrollable,
+}: Options = {}) {
+  return function addProvider<C>(
+    WrappedComponent: ReactComponent<OwnProps & WithAppProviderProps, C>,
+  ): React.ComponentClass<OwnProps> & C {
+    // eslint-disable-next-line react/prefer-stateless-function
+    class WithProvider extends React.Component<OwnProps, never> {
+      static contextTypes = WrappedComponent.contextTypes;
 
       render() {
         return (
-          <ThemeProviderConsumer>
-            {({polarisTheme}) => {
-              const {polaris} = this.context;
-              const polarisContext = {
-                ...polaris,
-                theme: polarisTheme,
-              };
+          <AppProviderContext.Consumer>
+            {(polaris) => {
+              return (
+                <ThemeProviderConsumer>
+                  {(polarisTheme) => {
+                    const polarisContext: PolarisContext = {
```

Comment:
We don't need the type addition here since we're covered with https://github.com/Shopify/polaris-react/pull/1424/files#diff-3013291a8b8e3e11eed7ce91df75f044R64 but it can't hurt